src/tools/sandbox.rs

```diff
@@ -1,68 +1,78 @@
+#[cfg(not(target_os = "macos"))]
 use std::path::Path;
 
 use anyhow::{Context, Error};
 use tabox::configuration::SandboxConfiguration;
 use tabox::syscall_filter::SyscallFilter;
 use tabox::{Sandbox, SandboxImplementation};
+
+#[cfg(not(target_os = "macos"))]
 use tempdir::TempDir;
 
+#[cfg(not(target_os = "macos"))]
 use task_maker_exec::sandbox::READABLE_DIRS;
 
 use crate::tools::opt::SandboxOpt;
 
 pub fn main_sandbox(opt: SandboxOpt) -> Result<(), Error> {
-    let tempdir = TempDir::new("tm-tools-sandbox")?;
-    let etcdir = tempdir.path();
-
     let mut config = SandboxConfiguration::default();
-    config.working_directory("/box");
-    let workdir = opt
-        .workdir
-        .or_else(|| std::env::current_dir().ok())
-        .unwrap_or_else(|| "/".into());
-    config.mount(&workdir, "/box", true);
-    config.env("PATH", std::env::var("PATH").unwrap_or_default());
-    config.env("LANG", std::env::var("LANG").unwrap_or_default());
-    config.env("LC_ALL", std::env::var("LC_ALL").unwrap_or_default());
 
-    config.mount(etcdir, "/etc", true);
-    std::fs::write(
-        etcdir.join("passwd"),
-        format!(
-            "root::0:0::/:/bin/sh\n\
-            nobody::{uid}:{gid}::/:/bin/sh\n",
-            uid = opt.uid,
-            gid = opt.gid,
-        ),
-    )
-    .with_context(|| format!("Failed to write /etc/etcdir in {}", etcdir.display()))?;
-    std::fs::write(
-        etcdir.join("group"),
-        format!(
-            "root:x:0:root\n\
-            nobody:x:{gid}:nobody\n",
-            gid = opt.gid,
-        ),
-    )
-    .with_context(|| format!("Failed to write /etc/group in {}", etcdir.display()))?;
-
-    for dir in READABLE_DIRS {
-        if Path::new(dir).is_dir() {
-            config.mount(dir, dir, false);
+    #[cfg(not(target_os = "macos"))]
+    {
+        let tempdir = TempDir::new("tm-tools-sandbox")?;
```

Comment:
This introduced a bug that somehow got unnoticed:

- here the directory is created
- the path of the directory is stored in the config
- at the end of this `#[cfg(not(target_os = "macos"))]` scope the directory is dropped (!!)
- the sandbox is actually started outside the scope, but the directory is already gone (!!!)

```
[2021-11-30T08:38:49.439982113Z ERROR tabox::linux] Child failed: Failed to create sandbox filesystem

Caused by:
    0: Failed to mount /tmp/tm-tools-sandbox.o3u0gReBD5vY -> /etc
    1: Failed to bind-mount /tmp/tm-tools-sandbox.o3u0gReBD5vY -> /tmp/tabox.bkkCC6uj52WV/etc
    2: ENOENT: No such file or directory
Error: Failed to wait sandbox

Caused by:
    Watcher thread failed

Caused by:
    Failed to create sandbox filesystem

Caused by:
    0: Failed to mount /tmp/tm-tools-sandbox.o3u0gReBD5vY -> /etc
    1: Failed to bind-mount /tmp/tm-tools-sandbox.o3u0gReBD5vY -> /tmp/tabox.bkkCC6uj52WV/etc
    2: ENOENT: No such file or directory
```

Reply:
Should be fixed in 8209c109f394b6c07208f24174260da6c2bac44b.